File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Food.java
package com.logistics.Components.Product;

public class Food extends Consumable {

    private boolean isCanned;

    // Constructors
    public Food() {
        super();
        isCanned = false;
    }

    public Food(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, int calories, boolean isCanned) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, calories);
        this.isCanned = isCanned;
    }

    // Copy Constructor
    public Food(Food other) {
        super(other); // Call the copy constructor of the superclass (Consumable)
        this.isCanned = other.isCanned;
    }

    // Clone method
    @Override
    public Food clone() {
        return new Food(this);
    }

    // Getters
    public boolean getIsCanned() { return isCanned; }

    // Setters
    public void setIsCanned(boolean isCanned) { this.isCanned = isCanned; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nCanned: " + isCanned;
    }
}

File: Task 1/C++/Components/Product/Food.cpp
#include "Food.h"

Food::Food() : Consumable(){
    bool isCanned = false;
}

Food::Food(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, int calories, bool isCanned)
    : Consumable(name, cost, supplier, discount_amount, packing_units, cost_per_unit, calories){
    this->isCanned = isCanned;
}

Food::Food(const Food& other) : Consumable(other) {
    this->isCanned = other.isCanned;
}

std::string Food::display_product_info() {
    return Consumable::display_product_info() 
    + "\nIs Canned: " + (isCanned? "true" : "false");
}